GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.cc

```diff
@@ -0,0 +1,95 @@
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "GeneratorInterface/GenFilters/plugins/MCLongLivedParticles.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include <iostream>
+#include <vector>
+
+using namespace edm;
+using namespace std;
+
+//Filter particles based on their minimum and/or maximum displacement on the transverse plane and optionally on their pdgIds
+//To run independently of pdgId, do not insert the particleIDs entry in filter declaration
+
+MCLongLivedParticles::MCLongLivedParticles(const edm::ParameterSet& iConfig)
+    : moduleLabel_(iConfig.getParameter<std::string>("hepMCProductTag")),
+      hepMCProductTag_(
+          edm::InputTag(iConfig.getUntrackedParameter(moduleLabel_, std::string("generator")), "unsmeared")),
+      token_(consumes<edm::HepMCProduct>(hepMCProductTag_)),
+      particleIDs(iConfig.getParameter<std::vector<int>>("ParticleIDs")),
+      theUpperCut_(iConfig.getParameter<double>("LengMax")),
+      theLowerCut_(iConfig.getParameter<double>("LengMin")) {}
+
+void MCLongLivedParticles::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("hepMCProductTag", "moduleLabel");
```

Comment:
```suggestion
  desc.add<std::string>("hepMCProductTag", "moduleTag");
```
if you want to use a generic name as default, perhaps something which includes "Tag" instead of "Lablìel" could be preferred
Even better if, instead of"moduleTag" you write the actual tag, e.g. (taking the same example as in your L19 above):
```C++
  desc.add<std::string>("hepMCProductTag", ("moduleLabel","generator","unsmeared));
```
(where in place of "moduleLabel" you should write the actual product that you want to get)